src/main/java/uk/gov/companieshouse/lookup/service/impl/CompanyLookupServiceImpl.java

```diff
@@ -1,41 +1,89 @@
 package uk.gov.companieshouse.lookup.service.impl;
 
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 import org.springframework.web.util.UriTemplate;
 import uk.gov.companieshouse.api.ApiClient;
 import uk.gov.companieshouse.api.error.ApiErrorResponseException;
 import uk.gov.companieshouse.api.handler.exception.URIValidationException;
 import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.model.company.RegisteredOfficeAddressApi;
+import uk.gov.companieshouse.api.model.company.account.CompanyAccountApi;
 import uk.gov.companieshouse.lookup.exception.ServiceException;
+import uk.gov.companieshouse.lookup.model.CompanyConfirmation;
+import uk.gov.companieshouse.lookup.model.CompanyLookup;
+import uk.gov.companieshouse.lookup.service.ApiClientService;
 import uk.gov.companieshouse.lookup.service.CompanyLookupService;
-import uk.gov.companieshouse.sdk.manager.ApiSdkManager;
+import uk.gov.companieshouse.lookup.validation.ValidationError;
 
 @Service
 public class CompanyLookupServiceImpl implements CompanyLookupService {
 
     private static final UriTemplate GET_COMPANY_URI =
         new UriTemplate("/company/{companyNumber}");
 
-    @Override
-    public CompanyProfileApi getCompanyProfile(String companyNumber) throws ServiceException {
-
-        ApiClient apiClient = ApiSdkManager.getSDK();
+    @Autowired
+    private ApiClientService apiClientService;
 
-        CompanyProfileApi companyProfileApi;
+    @Override
+    public CompanyConfirmation getCompanyProfile(String companyNumber)
+        throws ServiceException {
 
+        ApiClient apiClient = apiClientService.getApiClient();
         String uri = GET_COMPANY_URI.expand(companyNumber).toString();
 
         try {
-            companyProfileApi = apiClient.company().get(uri).execute();
+            return mapCompany(apiClient.company().get(uri).execute());
+        } catch (URIValidationException e) {
+            throw new ServiceException("Exception building URI", e);
         } catch (ApiErrorResponseException e) {
+            if (e.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
+                return null;
+            }
+            throw new ServiceException("API Error response", e);
```

Comment:
Again, can the message be made more meaningful to aid support? 'Failed to retrieve company profile for ' + companyNumber?